scripts/build/Dockerfile_klee

```diff
@@ -1,10 +1,24 @@
-# Build the final KLEE layer
-ARG REPOSITORY
-ARG LLVM_VERSION_SHORT
 ARG LLVM_SUFFIX
+ARG LLVM_VERSION_SHORT
+ARG REPOSITORY
 ARG SOLVER_SUFFIX
-ARG DEPS_SUFFIX
-FROM ${REPOSITORY}/klee_deps:${LLVM_VERSION_SHORT}${LLVM_SUFFIX}${SOLVER_SUFFIX}${DEPS_SUFFIX}
+ARG SANITIZER_SUFFIX
+FROM ${REPOSITORY}/klee_deps:${LLVM_VERSION_SHORT}${LLVM_SUFFIX}${SANITIZER_SUFFIX} as llvmdeps
+
+ARG REPOSITORY
+ARG SOLVER_SUFFIX
+FROM ${REPOSITORY}/solver:${SOLVER_SUFFIX}${SANITIZER_SUFFIX} as solver
+
+ARG BASE=/tmp
+#Remove all but the install dirs from the solver image, so that we can copy the whole
+#${BASE}/ dir, because we don't know which solvers are there.
+RUN bash -O extglob -c "rm -r !(*-install|metaSMT)"
+#Create the dir so COPY does not fail if this image has no metaSMT
+RUN mkdir -p ${BASE}/metaSMT
+
+# Build the final KLEE layer
+ARG REPOSITORY
+FROM ${REPOSITORY}/base
```

Comment:
After building all those things, can we use a slimmer base here?
There is a LLVM compiler in `PATH`, does a user need more to start experimenting with KLEE?

Reply:
We need most of those things to compile KLEE I thought? The base image isn't that big anyway.

Reply:
Sure, i was thinking about a user-only scenario, a user tests a software with KLEE.

Reply:
Then we can add a super-light image (ubuntu or alpine base) and copy just the needed libraries and binaries  over? That might be a very cool idea, but it's out of scope of this PR I think. This could also resolve the `klee` user issue, since the build image is root only and in the user-only scenario we can have the `klee` user?